hydro_cli/src/core/hydroflow_crate/build.rs

```diff
@@ -36,48 +32,63 @@ pub fn build_crate(
     };
     unit_of_work
         .get_or_init(|| {
-            let config = Config::default().unwrap();
-            config.shell().set_verbosity(cargo::core::Verbosity::Normal);
+            let mut build_args = vec!["build".to_string(), "--release".to_string()];
 
-            let workspace = Workspace::new(&src, &config).unwrap();
+            if let Some(example) = example.as_ref() {
+                build_args.push("--example".to_string());
+                build_args.push(example.clone());
+            }
 
-            let mut compile_options = CompileOptions::new(&config, CompileMode::Build).unwrap();
-            compile_options.filter = CompileFilter::Only {
-                all_targets: false,
-                lib: LibRule::Default,
-                bins: FilterRule::Just(vec![]),
-                examples: FilterRule::Just(vec![example.unwrap()]),
-                tests: FilterRule::Just(vec![]),
-                benches: FilterRule::Just(vec![]),
-            };
+            match target_type {
+                HostTargetType::Local => {}
+                HostTargetType::Linux => {
+                    build_args.push("--target".to_string());
+                    build_args.push("x86_64-unknown-linux-musl".to_string());
+                }
+            }
 
-            compile_options.build_config = BuildConfig::new(
-                &config,
-                None,
-                false,
-                &(match target_type {
-                    HostTargetType::Local => vec![],
-                    HostTargetType::Linux => vec!["x86_64-unknown-linux-musl".to_string()],
-                }),
-                CompileMode::Build,
-            )
-            .unwrap();
+            if let Some(features) = features {
+                build_args.push("--features".to_string());
+                build_args.push(features.join(","));
+            }
 
-            compile_options.build_config.requested_profile = InternedString::from("release");
-            compile_options.cli_features =
-                CliFeatures::from_command_line(&features.unwrap_or_default(), false, true).unwrap();
+            build_args.push("--message-format=json-render-diagnostics".to_string());
 
-            let res = cargo::ops::compile(&workspace, &compile_options).unwrap();
-            let binaries = res
-                .binaries
-                .iter()
-                .map(|b| b.path.to_string_lossy())
-                .collect::<Vec<_>>();
+            let mut command = Command::new("cargo")
```

Comment:
I guess you can avoid all the `.to_string()` disparity if you declare `let mut command = Command::new(...)` at the top, then do `command.arg("--features");` (instead of `build_args.push("--features".to_string())`)